packages/clay-core/stories/Picker.stories.tsx

```diff
@@ -0,0 +1,149 @@
+/**
+ * SPDX-FileCopyrightText: © 2022 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import Form from '@clayui/form';
+import Icon from '@clayui/icon';
+import Label from '@clayui/label';
+import Layout from '@clayui/layout';
+import {useId} from '@clayui/shared';
+import React from 'react';
+
+import {Option, Picker} from '../src/picker';
+import {Text} from '../src/typography';
+
+export default {
+	component: Picker,
+	title: 'Design System/Components/Picker',
+};
+
+export const Default = () => {
+	const pickerId = useId();
+	const labelId = useId();
+
+	return (
+		<div style={{width: '150px'}}>
+			<Form.Group>
+				<label htmlFor={pickerId} id={labelId}>
+					Choose a fruit
+				</label>
+				<Picker aria-labelledby={labelId} id={pickerId}>
+					<Option key="apple">Apple</Option>
+					<Option disabled key="banana">
+						Banana
+					</Option>
+					<Option key="mangos">Mangos</Option>
+					<Option key="blueberry">Blueberry</Option>
+					<Option key="boysenberry">Boysenberry</Option>
+					<Option key="cherry">Cherry</Option>
+					<Option key="cranberry">Cranberry</Option>
+					<Option key="eggplant">Eggplant</Option>
+					<Option key="fig">Fig</Option>
+					<Option key="grape">Grape</Option>
+					<Option key="guava">Guava</Option>
+					<Option key="huckleberry">Huckleberry</Option>
+				</Picker>
+			</Form.Group>
+		</div>
+	);
+};
+
+export const Dynamic = () => {
+	const pickerId = useId();
+	const labelId = useId();
+
+	return (
+		<div style={{width: '150px'}}>
+			<Form.Group>
+				<label htmlFor={pickerId} id={labelId}>
+					Choose a fruit
+				</label>
+				<Picker
+					aria-labelledby={labelId}
+					id={pickerId}
+					items={[
+						'Apple',
+						'Banana',
+						'Mangos',
+						'Blueberry',
+						'Boysenberry',
+						'Cherry',
+						'Cranberry',
+						'Eggplant',
+						'Fig',
+						'Grape',
+						'Guava',
+						'Huckleberry',
+					]}
+				>
+					{(item) => <Option key={item}>{item}</Option>}
+				</Picker>
+			</Form.Group>
+		</div>
+	);
+};
+
+const Trigger = React.forwardRef<
+	HTMLDivElement,
+	React.HTMLAttributes<HTMLDivElement>
+>(({children, ...otherProps}, ref) => (
```

Comment:
Trigger missing id and for attributes.

Reply:
It receives all the internal properties that the component passes to the trigger.

For example:

```jsx
<Picker
  aria-labelledby={labelId}
  as={Trigger}
  id={pickerId}
  items={['Liam', 'Noah', 'Oliver']}
>
  {...}
</Picker>
```

Internally we set all the properties for the trigger, so we pass `aria-activedescendant`, `aria-controls`, events and etc. everything that is needed. The `as` API works as a trigger replacement which by default is a `button`, so when you pass a component it passes all the properties that are defined in the trigger to your custom trigger.

```tsx
function Picker<T>({as: As = 'button', ...otherProps}: Props<T>) {
  return (
    <As
      {...otherProps}
      aria-activedescendant={active ? activeDescendant : ''}
      aria-controls={active ? ariaControls : undefined}
    >
      ...
    </As>
  )
}
```

This is an example of how this pattern works, so that your custom receives all the necessary properties to keep the component working and accessible.